candle-core/src/safetensors.rs

```diff
@@ -71,7 +71,7 @@ fn convert_<T: WithDType>(view: st::TensorView<'_>, device: &Device) -> Result<T
         let data: &[T] = unsafe { std::slice::from_raw_parts(v.as_ptr() as *const T, elem_count) };
         Tensor::from_slice(data, view.shape(), device)
     } else {
-        let mut c = Vec::with_capacity(elem_count);
+        let mut c: Vec<T> = Vec::with_capacity(elem_count);
```

Comment:
Could you add a comment to state the problem when this annotation is not there? This will avoid potentially removing it in further changes.

Reply:
Done. Is that explanation clear enough ?

Reply:
Looks great, thanks!